server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/KeywordsConsistencyOnDelegationMailbox.feature

```diff
@@ -0,0 +1,189 @@
+    Given a domain named "domain.tld"
+    And a user "alice@domain.tld"
+    And a user "bob@domain.tld"
+    And "alice@domain.tld" has a mailbox "notShared"
+    And "alice@domain.tld" has a mailbox "shared"
+    And "alice@domain.tld" shares its mailbox with write right "shared" with "bob@domain.tld"
+    And "alice@domain.tld" is connected
+    And the user has a message "mess" in "notShared" mailbox with subject "My awesome subject", content "This is the content"
+    And the user copy "mess" from mailbox "notShared" to mailbox "shared"
+
+  Scenario: getMessageList filtered by flag should combine flag when delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared,notShared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  Scenario: getMessageList filtered by flag should keep flag on delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "notShared" with flag "$Flagged"
+    Then the message list is empty
+
+  Scenario: getMessageList filtered by flag should keep flag on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  @Ignore
+  Scenario: (Does not work) Get message list should insection Draft flag on all mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared,notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailbox "notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailbox "shared" with flag "$Draft"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  Scenario: getMessage with shared user should return message with combine flag when delegation mailbox
```

Comment:
Where is the combination here ?

Reply:
We had message on two difference mailboxes (shared and notShared) 
We update flag on only on mailbox (shared)

The flag of return message is combined on all mailboxes that it belongs to. (for all flags except $Draft)